these-tests/test/Tests/Semialign.hs

```diff
@@ -76,8 +82,32 @@ alignProps = testGroup "Align"
     , semialignLaws  (CZip     :: CSemialign Identity)
     , semialignLaws  (CUnAll   :: CSemialign Proxy)
     , semialignLaws  (CZip     :: CSemialign (Tagged Char))
+    -- note: with e.g. N5 (which isn't distributive lattice) distributivity laws fail!
+    , semialignLaws  (CZip     :: CSemialign (Const M2))
     ]
 
+-------------------------------------------------------------------------------
+-- Const
+-------------------------------------------------------------------------------
+
+instance Lattice a => Semialign (Const a) where
+    alignWith _ (Const a) (Const b) = Const (a \/ b)
+
+-- This is valid when @a@ is distributive lattice
+-- otherwise distributivity laws don't hold.
+instance Lattice a => Zip (Const a) where
+    zipWith _ (Const a) (Const b) = Const (a /\ b)
+
+-- Note: idempotency of lattice makes this valid!
+instance Lattice a => Unzip (Const a) where
+    unzip (Const a) = (Const a, Const a)
```

Comment:
Hello. Is there ever a case where this definition isn't the same as `unzip :: f a -> (f a, f a)` always just `unzip' . (\x -> (x, x))` where `unzip' :: Functor f => f (a, b) -> (f a, f b); unzip' fab = (fmap fst fab, fmap snd fab)`?

Reply:
You have your types wrong, but as far as I understand, no.

There's default implementation https://hackage.haskell.org/package/semialign-1.1/docs/Data-Zip.html#v:unzipDefault (which is not used, as `unzip` and `unzipWith` default implementations are defined in terms of each other).

And e.g. list has single traversal implementation: 

```haskell
-- | 'unzip' transforms a list of pairs into a list of first components
-- and a list of second components.
unzip    :: [(a,b)] -> ([a],[b])
{-# INLINE unzip #-}
unzip    =  foldr (\(a,b) ~(as,bs) -> (a:as,b:bs)) ([],[])
```

https://hackage.haskell.org/package/base-4.12.0.0/docs/src/GHC.List.html#unzip